tabbycat/standings/teams.py

```diff
@@ -189,7 +189,7 @@ def annotate(self, queryset, standings, round=None):
         opponents_annotation = ArrayAgg('debateteam__debate__debateteam__team_id',
                 filter=opponents_filter)
         logger.info("Opponents annotation: %s", str(opponents_annotation))
-        teams_with_opponents = queryset.all().annotate(opponent_ids=opponents_annotation)
+        teams_with_opponents = queryset.model.objects.annotate(opponent_ids=opponents_annotation)
```

Comment:
Ah, that's a good idea. I guess the downside is that we would lose all the earlier filtering, if any, on `queryset`: This query would just return all teams in the database, right? Which is not too consequential, as subsequent lines only pull the relevant teams from it, but it's just suboptimal.

I was thinking ideally we'd have some sort of `distinct` in a subquery or something. But it wasn't obvious to me how to do this (and I haven't tried it myself). (I also thought about applying distinct to the annotation but of course teams can face each other twice so this wouldn't work.)